src/cmd/moby/run_qemu.go

```diff
@@ -73,18 +91,38 @@ func runQemu(args []string) {
 	} else {
 		qemuArgs = append(qemuArgs, "-enable-kvm")
 		qemuArgs = append(qemuArgs, "-machine", "q35,accel=kvm:tcg")
+		if container {
+			dockerArgs = append(dockerArgs, "", "")
+			copy(dockerArgs[5:], dockerArgs[3:])
```

Comment:
your patch patch sets up dockerArgs so why don't you set it up so that these gymnastics are not necessary? Your code looks a bot like:
```
a = 5
...
a = a + 3
a = a - 1
```
when it could look like:
```
a = 7
```